### PR TITLE
Add new cog which generates a markov chain "joke"

### PR DESCRIPTION
**Pull request by etcadinfinitum**
It's actually pretty silly, so we'll see if it works as intended.

Hi!

It looks like you are submitting a new pull request.

Please ensure that you've checked in all the files that you need for your project to build.
Your branch will be run through continuous integration to ensure code quality, however you 
should also provide proof that your code is doing what you say it does. Examples and screenshots
are good!

Keep in mind that all changes and PRs will have to go through the contributors first before being added.
It is completely up to our discretion what we add and what we don't add.

### CODE REVIEW

File: cogs/markov.py [hunk 1]
```diff
@@ -162,6 +163,25 @@ def predict(self, num_words: int, guild_id: int, user_id: int = None, start_word
...
+        else:
+            idx = random.randint(1, len(words) - 1)
+        words[idx] = words[idx] + "?\n"
+        words[len(words) - 1] = words[len(words) - 1] + "!"
+        return words
+
+    @commands.command("markov_joke")
+    @commands.cooldown(5, 30, commands.BucketType.user)
+    @commands.guild_only()
+    async def markov_joke(self, ctx, user: discord.User, words: int = 15):
```
Comment (Chris-Johnston) [inline cogs/markov.py]: do we want this based off of a specific user? I mean I guess that would be pretty funny. Doesn't really help that the spooky-related stuff is in a different project than the markov-related stuff.

Comment (etcadinfinitum) [reply cogs/markov.py]: > do we want this based off of a specific user?

That's a good question; I didn't put a ton of effort into understanding how the generalized chain is invoked differently than the user-specific chain.

> Doesn't really help that the spooky-related stuff is in a different project than the markov-related stuff.

Correct

Comment (Chris-Johnston) [reply cogs/markov.py]: I'd update this method to end with `_user`, just like the other methods. We can also add another method that doesn't require a user to be specified:
```py
    @commands.command("markov_joke")
    @commands.cooldown(5, 30, commands.BucketType.user)
    @commands.guild_only()
    async def markov_joke(self, ctx, words: int = 20):
        """
        Replies with a predicted phrase morphed into a 'joke' from the messages of all users.
        """
        async with ctx.channel.typing():
            await ctx.send(self.add_punctuation(self.predict(words, ctx.guild.id)))
```

**Review by Chris-Johnston — CHANGES_REQUESTED** [review]

File: cogs/markov.py (hunk 1 above)
Comment (etcadinfinitum) [reply cogs/markov.py]: See new commit with requested changes :+1:

**Review by Chris-Johnston — APPROVED** [review]
:shipit:

**Pull request merged**